src/vs/workbench/services/decorations/browser/decorationsService.ts

```diff
@@ -398,8 +398,8 @@ export class DecorationsService implements IDecorationsService {
 
 	private _keepItem(map: DecorationEntry, provider: IDecorationsProvider, uri: URI, data: IDecorationData | undefined): IDecorationData | null {
 		const deco = data ? data : null;
-		const old = map.set(provider, deco);
```

Comment:
`map.set` returns the map so `old` was always truthy

Reply:
Seems like here decorations will not fire if `data` is undefined. Is that intended?

Reply:
Fixed it